doc/ir-semantics.text

```diff
@@ -166,17 +175,20 @@ Update (partial) function, returns an H:
                if v = false then l₂
 
   step   osr(e, f^, lᵥ^, l^, (osr-map*)) (P, I, T, H, E, l)
-       = (I', T, H, E', l')
+       = (I', T, H', E', l')
     when v := eval P H E e
      and if v = false then
          I' := I
          l' := l
+         H' := H
          E' := E
      and if v = true then
          Df ∈ dom P
          Df  = Function f^ (formal*)
          I' := P(Df, lᵥ^)
          l' := l^
+         M  := (materialize P H E osr-map)*
+         H' := H ∪ rng(M)
          E' := (osr-env P H E osr-map)*
```

Comment:
`M` was added to `osr-env`, so this should be: `E' := (osr-env P H E M osr-map)*`